src/osd/PG.cc

```diff
@@ -324,6 +324,12 @@ bool PG::proc_replica_info(pg_shard_t from, const pg_info_t &oinfo)
     return false;
   }
 
+  if (get_osdmap()->is_down(from.osd)) {
```

Comment:
shouldn't this actually check is_down(from.osd) || get_up_from(from.osd) > sender's map_epoch?